pkg/generators/snippets/marshaljson.go

```diff
@@ -0,0 +1,30 @@
+package snippets
+
+import (
+	"errors"
+	"io"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/types"
+)
+
+func GenerateMarshalJSON(ctx *generator.Context, t *types.Type, embeddedTypePackage string, w io.Writer) error {
```

Comment:
I chose to have `embeddedTypePackage string` as an input parameter, and leave the finding and importing of the embedded type to the calling code.

The other way is to have imports as an input parameter. In that case, the calling code would need to `imports.AddType(embeddedType)`, and within this function we'd need to find the embeddedType from the  `t` type, and then do `"alias": imports.LocalNameOf(embeddedType.Name.Package)` on L18.

I think the current way is cleaner but open to suggestions.

Reply:
I that this is fine.  These functions should only implement the snippet writer.

My suggestion is finding a better name for `embeddedTypePackage`.   Maybe, `schemaGroupVersionPackageAlias`?